src/inventory_ui.cpp

```diff
@@ -910,7 +915,21 @@ item_location game::inv_map_splice(
                     }
                 }
             }
-            inv_s.make_item_list( slices.back(), pos == g->u.pos() ? &ground_cat : &nearby_cat );
```

Comment:
The deleted call to `inv_s.make_item_list` is made once per map tile. You can implement your category naming function here. The vehicle tiles are handled similarly.

> The loop is not used to find an item. It filters items and stack them when they are the same

How does this differ from the previous implementation?